src/cljdoc/server/search/search.clj

```diff
@@ -134,19 +141,19 @@
            tokens match-modes)
          (apply boolean-query BooleanClause$Occur/MUST))))
 
-(defn search->results [^String index-dir ^String query-in f]
-  ;; TODO If no matches, try again but using OR instead of AND in multi-tokens->query?
-  (with-open [reader (DirectoryReader/open (fsdir index-dir))]
-    (let [page-size        30
-          searcher         (IndexSearcher. reader)
-          ;parser           (QueryParser. "artifact-id" analyzer)
-          ;query            (.parse parser query-str)
-          ^Query query     (if (instance? Query query-in)
-                             query-in
-                             (parse-query query-in))
-          ^TopDocs topdocs (.search searcher query page-size)
-          hits             (.scoreDocs topdocs)]
-      (f {:topdocs topdocs :score-docs hits :searcher searcher :query query}))))
+(defn search->results
+  ([^String index-dir ^String query-in f] (search->results index-dir query-in 30 f))
+  ([^String index-dir ^String query-in ^long max-results f]
+   (with-open [reader (DirectoryReader/open (fsdir index-dir))]
+     (let [searcher         (IndexSearcher. reader)
+           ;parser           (QueryParser. "artifact-id" analyzer)
+           ;query            (.parse parser query-str)
+           ^Query query     (if (instance? Query query-in)
+                              query-in
+                              (parse-query query-in))
+           ^TopDocs topdocs (.search searcher query max-results)
+           hits             (.scoreDocs topdocs)]
+       (f {:topdocs topdocs :score-docs hits :searcher searcher :query query})))))
```

Comment:
This is slightly unrelated to the change of this PR but passing in `f` here seems odd to me. The difference in usage seems to be

```clj
(search->results ,,, f)
;; vs
(f (search->results ,,,))
```
I find the latter much clearer in terms of understanding how `f` is used. Passing it in makes that a little less obvious. My 2 cents anyway, feel free to ignore :)

Reply:
I have added a clarification comment. It needs to be that way because it must be run within the `search`'s `with-open`, it cannot run after the search finished.

Reply:
Ah interesting, didn’t consider this :)

Reply:
Fun fact: me too :) Coming back to the code to implement the suggest API, I looked at `search` and had the exact same thought as you. After factoring `f` out and calling it, I got the "index reader already closed" exception which reminded me of why did it this way in the first place. A good reminder that any code deviating from what is normal needs to be well explained :)